pandas/core/groupby/generic.py

```diff
@@ -434,15 +432,22 @@ def first_not_none(values):
                 # if we have date/time like in the original, then coerce dates
                 # as we are stacking can easily have object dtypes here
                 so = self._selected_obj
-                if (so.ndim == 2 and so.dtypes.apply(is_datetimelike).any()):
-                    result = result.apply(
-                        lambda x: to_numeric(x, errors='ignore'))
-                    date_cols = self._selected_obj.select_dtypes(
-                        include=['datetime', 'timedelta']).columns
-                    date_cols = date_cols.intersection(result.columns)
-                    result[date_cols] = (result[date_cols]
-                                         ._convert(datetime=True,
-                                                   coerce=True))
+                if so.ndim == 2 and so.dtypes.apply(is_datetimelike).any():
```

Comment:
can you factor this to a module level function (maybe start a groupby/conversion.py), was going to move the _try_cast things there as well

Reply:
Sure.